crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -24,7 +30,9 @@ pub(crate) struct TransactionBlockEffects {
     pub tx_data: Either<StoredTransaction, NativeTransactionData>,
 
     /// Deserialized representation of `stored.raw_effects`.
-    pub native: NativeTransactionEffects,
+    /// Or representation returned from the JSON-RPC server in case of tx execution.
+    /// TODO: remove this once we rework execution
+    pub effects: Either<NativeTransactionEffects, PreCommitTransactionBlockEffects>,
```

Comment:
`NativeTransactionEffects` is not easily fully derivable from `PreCommitTransactionBlockEffects` which we get from FN, so keeping both.

Reply:
Rather than doing this, seeing as we are making changes to JSON-RPC, can we add a `show_raw_effects` display option to get the bytes for the effect structure from JSON-RPC instead of the structured representation that we have to make do with here? It seems like we are trying to undo something that JSON-RPC is doing, and that we should just have it not do that in the first place.